lighthouse-core/gather/gatherers/scripts.js

```diff
@@ -20,21 +23,46 @@ class Scripts extends Gatherer {
   async afterPass(passContext, loadData) {
     const driver = passContext.driver;
 
-    /** @type {Object<string, string>} */
-    const scriptContentMap = {};
+    /** @type {LH.Artifacts['Scripts']} */
+    const scripts = [];
+
+    /** @type {string[]} */
+    const inlineScripts = await driver.evaluateAsync(`(() => {
+      ${getElementsInDocumentString};
+
+      return getElementsInDocument('script')
+        .filter(script => !script.src && script.text.trim())
+        .map(script => script.text);
+    })()`, {useIsolation: true});
+
+    if (inlineScripts.length) {
+      const mainResource = loadData.networkRecords.find(
+        request => URL.equalWithExcludedFragments(request.url, passContext.url));
```

Comment:
done. after the two merge PRs merge maybe we should just move that check into `URL.equalWithExcludedFragments` as an early return

Reply:
> maybe we should just move that check into URL.equalWithExcludedFragments as an early return

yeah it's a *bit* different to do generically because in these cases we know for a fact there are no fragments on the network records but agreed we should not have to do these things, the URL.* methods should just be fast by default :)